scripts/eosio_uninstall.sh

```diff
@@ -1,89 +1,125 @@
 #!/usr/bin/env bash
 set -eo pipefail
+VERSION=2.1
+# User input handling
+DEP_PROCEED=false
+FULL=false
+
+function usage() {
+   printf "Usage: $0 OPTION...
+  -i DIR      Directory where eosio is installed)
+  -y          Noninteractive mode (answers yes to every prompt)
+  -f          Removal of data directory (be absolutely sure you want to delete it before using this!)
+   \\n" "$0" 1>&2
+   exit 1
+}
+
+TIME_BEGIN=$( date -u +%s )
+if [ $# -ne 0 ]; then
+   while getopts "i:yf" opt; do
+      case "${opt}" in
+         i )
+            INSTALL_LOCATION=$OPTARG
+         ;;
+         y )
+            NONINTERACTIVE=true
+            PROCEED=true
+            DEP_PROCEED=true
+         ;;
+         f )
+            FULL=true
+         ;;
+         h )
+            usage
+         ;;
+         ? )
+            echo "Invalid Option!" 1>&2
+            usage
+         ;;
+         : )
+            echo "Invalid Option: -${OPTARG} requires an argument." 1>&2
+            usage
+         ;;
+         * )
+            usage
+         ;;
+      esac
+   done
+fi
+
+export CURRENT_WORKING_DIR=$(pwd) # relative path support
+
+# Ensure we're in the repo root and not inside of scripts
+cd $( dirname "${BASH_SOURCE[0]}" )/..
 
 # Load bash script helper functions
 . ./scripts/helpers/eosio.sh
 
-usage() {
-   printf "Usage --- \\n $ %s [ --full ] [ --force ]\\n
-     --full: Removal of data directory (be absolutely sure you want to delete it before using this!)\\n
-     --force: Unattended uninstall which works regardless of the eosio directory existence.\\n              This helps cleanup dependencies and start fresh if you need to.
-   \\n" "$0"
-}
+# Support relative paths : https://github.com/EOSIO/eos/issues/7560
+( [[ ! -z $INSTALL_LOCATION ]] && [[ ! $INSTALL_LOCATION =~ ^\/ ]] ) && export INSTALL_LOCATION="${CURRENT_WORKING_DIR}/$INSTALL_LOCATION"
 
 INSTALL_PATHS=()
 
-# User input handling
-PROCEED=false
-DEP_PROCEED=false
-FORCED=false
-FULL=false
-if [[ $@ =~ [[:space:]]?--force[[:space:]]? ]]; then
-   echo "[Forcing Unattended Removal: Enabled]"
-   FORCED=true
-   PROCEED=true
-   DEP_PROCEED=true
+if [[ $NONINTERACTIVE == false ]] && $FULL; then
+   while true; do
+      read -p "By specifying -f, removal of the eosio data directory will require a resync of data which can take days. Do you wish to proceed? (y/n) " PROCEED
+      case $PROCEED in
+         "" ) echo "What would you like to do?";;
+         0 | true | [Yy]* ) break;;
+         1 | false | [Nn]* ) exit 0;;
+         * ) echo "Please type 'y' for yes or 'n' for no.";;
+      esac
+   done
+elif $NONINTERACTIVE && $FULL; then
+   echo "${COLOR_RED}By specifying -f, removal of the eosio data directory will require a resync of data which can take days. The script is sleeping for 15 seconds to give you a chance to cancel if this is not what you want.${COLOR_NC}"
```

Comment:
If the user has specified `-y` for non-interactive (which is not the default), I don't think we should be waiting in the script for a user action that is likely not going to come.